pkg/archive/archive.go

```diff
@@ -225,6 +225,60 @@ func CompressStream(dest io.Writer, compression Compression) (io.WriteCloser, er
 	}
 }
 
+// ReplaceFileTarWrapper converts inputTarStream to a new tar stream
+// with a single file called header.Name with contents of file.
```

Comment:
The comment is a bit unclear. It makes it sound like the contents of the existing tar is replaced with just one file.